app/models/use_cases/create_diagnosis.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module UseCases
+  class CreateDiagnosis
+    class << self
+      def create_with_params(params)
+        diagnosis = Diagnosis.create visit_id: params['visit_id']
+        return nil unless params['diagnosed_needs']
```

Comment:
Is it relevant to create a diagnosis if there is no diagnosed_needs? (ie, move this line above)